### DISCUSSION TITLE
Movable chart separator

### DISCUSSION THREAD

**Discussion by Tokyo479** [jfree/jfreechart#408] [Q&A]
Has anyone been able to create a subchart which has a movable (with mouse) dividing border . it looks like it only can be fixed. Any comments / insights much appreciated.

![jfreechart](https://github.com/jfree/jfreechart/assets/174390156/d87053c5-a07b-49a9-a45c-0050f007b62f)

**Comment by Tokyo479** [comment]
Any advice on this appreciated....Thanks.

**Comment by trashgod** [comment]
Assuming an XY plot, you might try adjusting the _weight_ properties of the subplots  in your chosen listener. As I recall, 1:1 is equal-sized.

**Reply by Tokyo479** [reply]
I want to allow the user to be able to dynamically drag the divider with a cursor to resize the area depending on what the chart (bottom or top) they want to explore ... how is this done with weight properties ?

**Reply by trashgod** [reply]
This sounds feasible. You can override `mouseDragged()` like [this](https://stackoverflow.com/a/7101410/230513). Alternatively, you might also look  at a slider or spinner change listener. See the `calculateAxisSpace()` method  of your chosen combined plot to see how it apportions the total weight. Also, don't overlook the pan and zoom controls mentioned [here](https://stackoverflow.com/a/44151032/230513).

**Reply by Tokyo479** [reply]
understood. thanks!

**Comment by trashgod** [comment]
For reference, the fragment below illustrates an `Action` that toggles the weights of two subplots in a combined category plot between _n:m_ and _n-1:m+1_. For example, given an original subplot weight ratio of 2:1, it alternates with a ratio of 1:2.

```
var chart = createChart();
var content = new JPanel(new BorderLayout());
var control = new JPanel();
control.add(new JToggleButton(new AbstractAction("+/-") {
    @Override
    public void actionPerformed(ActionEvent e) {
        if (chart.getPlot() instanceof CombinedDomainCategoryPlot plot) {
            var list = plot.getSubplots();
            if (list.get(0) instanceof CategoryPlot sub0
                && list.get(1) instanceof CategoryPlot sub1) {
                if (e.getSource() instanceof JToggleButton button) {
                    if (button.isSelected()) {
                        sub0.setWeight(sub0.getWeight() - 1);
                        sub1.setWeight(sub1.getWeight() + 1);
                    } else {
                        sub0.setWeight(sub0.getWeight() + 1);
                        sub1.setWeight(sub1.getWeight() - 1);
                    }
                }
            }
        }
    }
}));
content.add(new ChartPanel(chart));
content.add(control, BorderLayout.SOUTH);
return content;
```
